### PR TITLE
37 forgotten password

### PR DESCRIPTION
**Pull request by Idontker**
RM: @valentinBraeutigam

### CODE REVIEW

**Review by valentinBraeutigam — COMMENTED** [review]
grafik
It would maybe be better to use the same window layout as the other Popup windows. Although the alert function is nice I think it would be better to have the same style. 

It is not possible for me to click the login button anymore. I had to add the (click) property to the 'Login' button again. Did it work for you?

I can't really test wether the link is set right since it links to the device and not to localhost with the right port. This is because we don't have a normal internet adress. I hope it will be set right when we have one.

Maybe you can move the 'Change Password' link somewhere else already (e.g. upper right corner of the main page)

File: Documentation/api/openapi_didentity.yaml [hunk 1]
```diff
@@ -88,6 +88,27 @@ paths:
         500:
           description: Internal server error.
 
+  /auth/password/forgot:
+    post:
+      tags:
+        - authentification-controller
```
Comment (valentinBraeutigam) [inline Documentation/api/openapi_didentity.yaml:94]: authentication as usual :D

File: src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/services/MailService.java [hunk 2]
```diff
@@ -51,7 +51,19 @@ public boolean sendInvitation(String to, String invitationLink) {
...
-    public boolean sendPassword(String to, String strongPassword) {
+    public boolean sendInitialPassword(String to, String strongPassword) {
+        String subject = "Initiales passwort für DIdentity";
+        String header = "Hier ist ihr initiales Passwort für ihren Login in der DIdentity App";
+        return sendPassword(subject, header, to, strongPassword);
+    }
+
+    public boolean sendNewPassword(String to, String strongPassword) {
+        String subject = "Neues Passwort für DIdentity";
+        String header = "Hier ist ihr neues automatisch geniertes Passwort für ihren Login in der DIdentity App";
```
Comment (valentinBraeutigam) [inline src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/services/MailService.java:62]: geniertes?

File: src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/services/MailService.java [hunk 3]
```diff
@@ -64,8 +76,8 @@ public boolean sendPassword(String to, String strongPassword) {
                     + "old_password=" + strongPassword;
 
             String htmlText = "<img src='cid:logo' alt='logo' height='200'> " +
```
Comment (valentinBraeutigam) [inline src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/services/MailService.java:78]: Du setzt in den anderen Methode extra den Text, aber verwendest ihn hier nicht. Im Text hier: DIDentity => zweites d bitte klein

File: src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/services/AuthenticationServiceTest.java [hunk 4]
```diff
@@ -168,4 +178,101 @@ void testLoginWithEmptyPassword() {
...
+        // -- then --
+        assertEquals(HttpStatus.valueOf(500), response.getStatusCode());
+    }
+
+    // TODO
+    @Test
+    void handleForgotPassword_shouldResetPasswordAndSendMail_Return200() {
+        // given
+        User user = UserSamples.getSampleUser();
+        String pass = "secure passwort";
```
Comment (valentinBraeutigam) [inline src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/services/AuthenticationServiceTest.java:218]: passwort?  :P

File: src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/services/AuthenticationServiceTest.java [hunk 5]
```diff
@@ -168,4 +178,101 @@ void testLoginWithEmptyPassword() {
...
+        assertEquals(pass, passwordCaptor.getValue());
+
+    }
+
+    // TODO
+    @Test
+    void handleForgotPassword_shouldReturn500OnMailserviceFail() {
+        // given
+        User user = UserSamples.getSampleUser();
+        String pass = "secure passwort";
```
Comment (valentinBraeutigam) [inline src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/services/AuthenticationServiceTest.java:250]: hier auch nochmal passwort?

File: src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.ts [hunk 6]
```diff
@@ -93,4 +95,9 @@ export class LoginPageComponent implements OnInit {
       },
     });
   }
+
+  openForgotPassword() {
+    this.dialogRef.open(ForgotPasswordPopUpComponent, {});
+    console.log('open done');
```
Comment (valentinBraeutigam) [inline src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.ts:101]: only in DevMode?

File: src/digitalIdentity-frontend/src/app/shared/pop-up/forgot-password-pop-up/forgot-password-pop-up.component.html [hunk 7]
```diff
@@ -0,0 +1,34 @@
+<div id="forgot-password">
+  <div class="headers">
+    <h2 class="edit-card-title">Forgot Password?</h2>
+    <h3>
+      Enter your email to reset your password. You will recive a mail containing
```
Comment (valentinBraeutigam) [inline src/digitalIdentity-frontend/src/app/shared/pop-up/forgot-password-pop-up/forgot-password-pop-up.component.html:5]: recieve

**Review by valentinBraeutigam — CHANGES_REQUESTED** [review]
- Please be careful about spelling mistakes (in the code but more important in the parts shown to a user (e.g. email))
- Please change or respond to the comments above

File: src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/services/MailService.java (hunk 3 above)
Comment (Idontker) [reply src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/services/MailService.java:78]: fixed

File: src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/services/AuthenticationServiceTest.java (hunk 4 above)
Comment (Idontker) [reply src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/services/AuthenticationServiceTest.java:218]: fixed

File: src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/services/AuthenticationServiceTest.java (hunk 5 above)
Comment (Idontker) [reply src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/services/AuthenticationServiceTest.java:250]: fixed

File: src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.ts [hunk 8]
```diff
@@ -16,14 +17,15 @@ export class LoginPageComponent implements OnInit {
...
-    private dialogRef: MatDialog,
+    public dialogRef: MatDialog,
     private router: Router,
     private HttpService: BackendHttpService,
     private route: ActivatedRoute
   ) {}
 
   ngOnInit(): void {
     this.route.queryParams.subscribe((params: Params) => {
+      console.log('Debug', params);
```
[the remark this answers is not in the document]
Comment (Idontker) [reply src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.ts:28]: forgot to remove this line ^^"

File: src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.ts (hunk 6 above)
Comment (Idontker) [reply src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.ts:101]: forgot to remove this line ^^"

**Review by valentinBraeutigam — APPROVED** [review]

**Pull request merged**